From d927d173cc0dc1a83ae71aa8661704d0dd77c131 Mon Sep 17 00:00:00 2001
From: Holger Macht <hmacht@suse.de>
Date: Fri, 9 Apr 2010 15:12:21 +0200
Subject: [PATCH 26/33] workaround for HP ProBook 5310 BIOS issue

Workaround HP ProBook 5310 BIOS issue by marking the scan code
generated by FN-F7 as reserved (bnc#568988)
---
 fdi/information/10freedesktop/30-keymap-hp.fdi |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/fdi/information/10freedesktop/30-keymap-hp.fdi b/fdi/information/10freedesktop/30-keymap-hp.fdi
index fe60671..88a4690 100644
--- a/fdi/information/10freedesktop/30-keymap-hp.fdi
+++ b/fdi/information/10freedesktop/30-keymap-hp.fdi
@@ -75,13 +75,18 @@
             <append key="input.keymap.data" type="strlist">e003:email</append>	<!-- envelope button -->
           </match>
 
+	  <match key="/org/freedesktop/Hal/devices/computer:system.hardware.product" contains="5310">
+            <append key="input.keymap.data" type="strlist">e021:reserved</append>
+          </match>
+
 	  <match key="/org/freedesktop/Hal/devices/computer:system.board.vendor" string="Quanta">
 	    <!-- HP Pavillion dv6315ea with empty system.hardware.* keys fd.o#19936 -->
 	    <match key="/org/freedesktop/Hal/devices/computer:system.board.product" string="30B7">
 	      <match key="/org/freedesktop/Hal/devices/computer:system.board.version" string="65.2B">
+                <remove key="input.keymap.data" type="strlist">e008:presentation</remove>
                 <append key="input.keymap.data" type="strlist">e008:media</append> <!-- FIXME: quick play -->
 	      </match>
-	     </match>
+	    </match>
 	  </match>
 
       	</match>
-- 
1.7.1

